packages/server-wallet/src/engine/engine.ts

```diff
@@ -1027,14 +993,25 @@ export class SingleThreadedEngine
     response.succeededObjectives.map(o => this.emit('objectiveSucceeded', o));
   }
 
+  /**
+   * Update the engine's knowledge about the funding for a channel.
+   *
+   * @param args - An object specifying the channelId, asset holder address and amount.
+   * @returns A promise that resolves to a channel output.
+   */
   // ChainEventSubscriberInterface implementation
-  async holdingUpdated({channelId, amount, assetHolderAddress}: HoldingUpdatedArg): Promise<void> {
-    const response = EngineResponse.initialize();
-
-    await this.store.updateFunding(channelId, BN.from(amount), assetHolderAddress);
+  async holdingUpdated(
+    {channelId, amount, assetHolderAddress}: HoldingUpdatedArg,
+    response = EngineResponse.initialize()
+  ): Promise<SingleChannelOutput> {
+    await this.store.updateFunding(channelId, amount, assetHolderAddress);
     await this.takeActions([channelId], response);
 
     response.channelUpdatedEvents().forEach(event => this.emit('channelUpdated', event.value));
+
+    // holdingUpdated may be called by updateChannelsForFunding, which therefore includes
```

Comment:
I would be in favor of removing `updateChannelsForFunding` to reduce cognitive overhead of interpreting the wallet engine api.

Reply:
I agree, but was reluctant to make that change in this PR!

Reply:
#3443